### ISSUE TITLE
Determine which products support checkout URLs

### ISSUE THREAD

**Issue by helgatheviking** [facebookincubator/facebook-for-woocommerce#1192]
Products that require configuration are not compatible with Facebook checkout URLs. (This includes grouped products, mix and match products, product bundles, composite products, name your price products, product add-ons, etc). This is a known issue (#145) and I understand that FB might not want to add compatibility for custom plugin out there. 

The given solution is to _universally_ disable checkout URLs and send customers to the product page, for every product, _regardless_ of whether the product supports being added to the cart or not. 

I think this is overkill as WooCommerce itself knows whether a product can be added automatically 

This is how WooCommerce decides which link to display in the shop archive:

```
if( $product->supports( 'ajax_add_to_cart' ) && $product->is_purchasable() && $product->is_in_stock() )
```

If the condition is true then it shows an ajaxified "Add to cart" link. If not, the shop archive shows a "customize" link that directs the user to the product page whether the product can be customized. 

I propose that Facebook should do the same on their end.  The benefit of doing so would be that stores could still use Checkout URLs where able, but send users to the product page when required instead of sending them to the cart where they will get errors.

NB: Not sure how you are syncing the data, but assume it's via the WooCommerce API? That might need an update to include the `supports` info in the product response.

**Comment by lgfogle** [comment]
Hey there! 👋

Lindsey here, from the SkyVerge product team! We recently took over development and support for the Facebook for WooCommerce extension.

In our recent 1.10 releases, we've fixed many issues impacting the store connection, pixel tracking, and product sync. Could you please update to the latest version of Facebook for WooCommerce and see if that resolves this issue?

If not, could you please [contact our support team](https://woocommerce.com/my-account/create-a-ticket/) or open a new issue here? We'd be happy to help troubleshoot further with you.

Thank you!

**Comment by helgatheviking** [comment]
@lgfogle hi Lindsey! Glad to see SkyVerge taking over this. I'm pretty sure this issue is in the Facebook side so it's hard for me to test. Was this specifically something y'all worked on?

**Comment by lgfogle** [comment]
Hey @helgatheviking! Thanks for bringing this one up again 👍 

We started working on this one, but our developers found that a one-size-fits-all approach wouldn't work to cover the different types of products impacted by these issues (e.g., Name Your Price, Grouped products, Mix and Match, Bundles, etc.), so we're taking it back to the drawing board so we can work through each product type individually. It's still on our roadmap, but we've got a few other projects to complete first before starting to tackle those issues.

**Comment by helgatheviking** [comment]
So my `supports('ajax_add_to_cart')` idea didn't work out? bummer.

**Comment by helgatheviking** [comment]
Hi @lgfogle bringing this back as Woo has [`has_options()`](https://github.com/woocommerce/woocommerce/blob/master/includes/abstracts/abstract-wc-product.php#L1740-L1742) method now for each product. And they recently merged my PR to add a filter there too.

If this were added to the REST API, it seems like something that could the be tested for on the Facebook side. If `has_options()` is true, go to the product page. If not, add it to cart.
